torchaudio/transforms/_transforms.py

```diff
@@ -1242,6 +1241,61 @@ def __init__(self, time_mask_param: int, iid_masks: bool = False, p: float = 1.0
             raise ValueError(f"The value of p must be between 0.0 and 1.0 ({p} given).")
         super(TimeMasking, self).__init__(time_mask_param, 2, iid_masks, p=p)
 
+class SpecAugment(torch.nn.Module):
+    r"""Apply time and frequency masking to a spectrogram.
+
+    Args:
+        n_time_masks (int): Number of time masks.
+        time_mask_param (int): Maximum possible length of the time mask.
+        n_freq_masks (int): Number of frequency masks.
+        freq_mask_param (int): Maximum possible length of the frequency mask.
+        iid_masks (bool): Applies iid masks to each of the examples in the batch dimension.
+            This option is applicable only when the input tensor is 4D.
+        p (float, optional): maximum proportion of time steps that can be masked.
+            Must be within range [0.0, 1.0]. (Default: 1.0)
+        zero_masking (bool): If True, use 0 as the mask value with 0, else use mean of the input tensor.
+    """
+    __constants__ = ["mask_param", "axis", "iid_masks", "p"]
+
+    def __init__(self, n_time_masks: int, time_mask_param: int, n_freq_masks: int, freq_mask_param: int, iid_masks: bool = True, p: float = 1.0, zero_masking: bool = False) -> None:
+
+        super(SpecAugment, self).__init__()
+        self.zero_masking = zero_masking
+        self.n_time_masks = n_time_masks
+        self.time_mask_param = time_mask_param
+        self.n_freq_masks = n_freq_masks
+        self.freq_mask_param = freq_mask_param
+        self.iid_masks = iid_masks
+        self.p = p
+        self.zero_masking = zero_masking
+
+    def forward(self, specgram: Tensor) -> Tensor:
```

Comment:
The pre-existing issue with the conventional making implementation is that it won't distinguish the padded frames and valid frames when batching samples. What do you think? We could extend (or add a new implementation) that performs masking in padding-aware manner.

Reply:
oh it's a good point. I thought about that and decided to leave it for later, since it'll make the function much more complicated and the effect should be minimal as we're already making sure utterances have similar lengths when batching data, so that padding areas are small